internal/services/mssqlmanagedinstance/mssql_managed_instance_failover_group_resource.go

```diff
@@ -380,6 +393,10 @@ func (r MsSqlManagedInstanceFailoverGroupResource) Read() sdk.ResourceFunc {
 						}
 					}
 
+					if secondaryType := props.SecondaryType; secondaryType != nil {
+						model.SecondaryType = string(pointer.From(props.SecondaryType))
+					}
```

Comment:
Since `pointer.From` handles the nil check for us, I think we can replace this with:
```suggestion
					model.SecondaryType = string(pointer.From(props.SecondaryType))
```